### ISSUE TITLE
An exception while creating "Magento\Setup\Console\Command\ModuleUninstallCommand"; no instance returned

### ISSUE THREAD

**Issue by milanstojanov** [magento/magento2#1716]
While trying to install Magento_SampleInterception module (https://github.com/magento/magento2-samples/tree/master/sample-module-interception) with the standard flow:

1. bin/magento module:enable Magento_SampleInterception
2. bin/magento setup:upgrade

I got the following exception:

Autoload error: An abstract factory could not create an instance of magentosetupconsolecommandmoduleuninstallcommand(alias: Magento\Setup\Console\Command\ModuleUninstallCommand).

After digging around for a while, I managed to get a bit more friendlier message:

An exception was raised while creating "Magento\Setup\Console\Command\ModuleUninstallCommand"; no instance returned.

However, it's still not very clear how and why it ended up here. 

As a temporary workaround, I manually added:

'Magento_SampleInterception' => 1 to etc/config.php 
and a new row in setup_module table: Magento_SampleInterception | 2.0.0 | 2.0.0

This did a trick, but an investigation is clearly needed.

P.S. All sample modules are requiring "magento/framework": "~0.74", which should be updated as well.

Thanks, 
Milan

**Comment by mazhalai** [comment]
@milanstojanov  in the `develop` branch we have pushed a fix that will give a more detailed error. Can you please try with the `develop` branch for more details?

**Comment by milanstojanov** [comment]
Hi, 

I pulled the latest code, and it's displaying the exceptions which I already tracked:

An abstract factory could not create an instance of magentosetupconsolecommandmoduleuninstallcommand(alias: Magento\Setup\Console\Command\ModuleUninstallCommand).#0 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(646): Zend\ServiceManager\ServiceManager->createFromAbstractFactory('magentosetupcon...', 'Magento\Setup\C...')
#1 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(598): Zend\ServiceManager\ServiceManager->doCreate('Magento\Setup\C...', 'magentosetupcon...')
#2 /var/www/html/magento2/setup/src/Magento/Setup/Console/CommandList.php(89): Zend\ServiceManager\ServiceManager->create('Magento\Setup\C...')
#3 /var/www/html/magento2/lib/internal/Magento/Framework/Console/Cli.php(56): Magento\Setup\Console\CommandList->getCommands()
#4 /var/www/html/magento2/lib/internal/Magento/Framework/Console/Cli.php(30): Magento\Framework\Console\Cli->getApplicationCommands()
#5 /var/www/html/magento2/vendor/symfony/console/Symfony/Component/Console/Application.php(91): Magento\Framework\Console\Cli->getDefaultCommands()
#6 /var/www/html/magento2/bin/magento(24): Symfony\Component\Console\Application->__construct('Magento CLI', '1.0.0-beta')
#7 {main}

An exception was raised while creating "Magento\Setup\Console\Command\ModuleUninstallCommand"; no instance returned#0 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(1129): Zend\ServiceManager\ServiceManager->createServiceViaCallback(Array, 'magentosetupcon...', 'Magento\Setup\C...')
#1 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(646): Zend\ServiceManager\ServiceManager->createFromAbstractFactory('magentosetupcon...', 'Magento\Setup\C...')
#2 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(598): Zend\ServiceManager\ServiceManager->doCreate('Magento\Setup\C...', 'magentosetupcon...')
#3 /var/www/html/magento2/setup/src/Magento/Setup/Console/CommandList.php(89): Zend\ServiceManager\ServiceManager->create('Magento\Setup\C...')
#4 /var/www/html/magento2/lib/internal/Magento/Framework/Console/Cli.php(56): Magento\Setup\Console\CommandList->getCommands()
#5 /var/www/html/magento2/lib/internal/Magento/Framework/Console/Cli.php(30): Magento\Framework\Console\Cli->getApplicationCommands()
#6 /var/www/html/magento2/vendor/symfony/console/Symfony/Component/Console/Application.php(91): Magento\Framework\Console\Cli->getDefaultCommands()
#7 /var/www/html/magento2/bin/magento(24): Symfony\Component\Console\Application->__construct('Magento CLI', '1.0.0-beta')
#8 {main}

Decoding failed: Syntax error#0 /var/www/html/magento2/lib/internal/Magento/Framework/Module/PackageInfo.php(91): Zend_Json::decode('{?  "name": "ma...')
#1 /var/www/html/magento2/lib/internal/Magento/Framework/Module/PackageInfo.php(156): Magento\Framework\Module\PackageInfo->load()
#2 /var/www/html/magento2/lib/internal/Magento/Framework/Module/DependencyChecker.php(124): Magento\Framework\Module\PackageInfo->getRequire('Magento_Store')
#3 /var/www/html/magento2/lib/internal/Magento/Framework/Module/DependencyChecker.php(48): Magento\Framework\Module\DependencyChecker->createGraph(Object(Magento\Framework\Module\PackageInfo))
#4 /var/www/html/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php(101): Magento\Framework\Module\DependencyChecker->__construct(Object(Magento\Framework\Module\ModuleList), Object(Magento\Framework\Module\ModuleList\Loader), Object(Magento\Framework\Module\PackageInfoFactory))
#5 /var/www/html/magento2/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php(89): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('Magento\Framewo...', Array)
#6 /var/www/html/magento2/lib/internal/Magento/Framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\Framewo...')
#7 /var/www/html/magento2/setup/src/Magento/Setup/Console/Command/ModuleUninstallCommand.php(160): Magento\Framework\ObjectManager\ObjectManager->get('Magento\Framewo...')
#8 [internal function]: Magento\Setup\Console\Command\ModuleUninstallCommand->__construct(Object(Magento\Framework\Composer\ComposerInformation), Object(Magento\Framework\App\DeploymentConfig), Object(Magento\Framework\App\DeploymentConfig\Writer), Object(Magento\Framework\Module\FullModuleList), Object(Magento\Framework\Module\ModuleList\Loader), Object(Magento\Framework\App\MaintenanceMode), Object(Magento\Setup\Model\ObjectManagerProvider), Object(Magento\Framework\Composer\Remove), Object(Magento\Setup\Model\UninstallCollector))
#9 /var/www/html/magento2/vendor/zendframework/zend-di/src/Di.php(496): ReflectionClass->newInstanceArgs(Array)
#10 /var/www/html/magento2/vendor/zendframework/zend-di/src/Di.php(299): Zend\Di\Di->createInstanceViaConstructor('Magento\Setup\C...', Array, NULL)
#11 /var/www/html/magento2/vendor/zendframework/zend-di/src/Di.php(248): Zend\Di\Di->newInstance('Magento\Setup\C...', Array, true)
#12 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/Di/DiServiceFactory.php(103): Zend\Di\Di->get('Magento\Setup\C...', Array)
#13 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/Di/DiAbstractServiceFactory.php(43): Zend\ServiceManager\Di\DiServiceFactory->get('Magento\Setup\C...', Array)
#14 [internal function]: Zend\ServiceManager\Di\DiAbstractServiceFactory->createServiceWithName(Object(Zend\ServiceManager\ServiceManager), 'magentosetupcon...', 'Magento\Setup\C...')
#15 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(939): call_user_func(Array, Object(Zend\ServiceManager\ServiceManager), 'magentosetupcon...', 'Magento\Setup\C...')
#16 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(1129): Zend\ServiceManager\ServiceManager->createServiceViaCallback(Array, 'magentosetupcon...', 'Magento\Setup\C...')
#17 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(646): Zend\ServiceManager\ServiceManager->createFromAbstractFactory('magentosetupcon...', 'Magento\Setup\C...')
#18 /var/www/html/magento2/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(598): Zend\ServiceManager\ServiceManager->doCreate('Magento\Setup\C...', 'magentosetupcon...')
#19 /var/www/html/magento2/setup/src/Magento/Setup/Console/CommandList.php(89): Zend\ServiceManager\ServiceManager->create('Magento\Setup\C...')
#20 /var/www/html/magento2/lib/internal/Magento/Framework/Console/Cli.php(56): Magento\Setup\Console\CommandList->getCommands()
#21 /var/www/html/magento2/lib/internal/Magento/Framework/Console/Cli.php(30): Magento\Framework\Console\Cli->getApplicationCommands()
#22 /var/www/html/magento2/vendor/symfony/console/Symfony/Component/Console/Application.php(91): Magento\Framework\Console\Cli->getDefaultCommands()
#23 /var/www/html/magento2/bin/magento(24): Symfony\Component\Console\Application->__construct('Magento CLI', '1.0.0-beta')
#24 {main}

**Comment by buskamuza** [comment]
```
Decoding failed: Syntax error#0 /var/www/html/magento2/lib/internal/Magento/Framework/Module/PackageInfo.php(91): Zend_Json::decode('{? "name": "ma...')
#1 /var/www/html/magento2/lib/internal/Magento/Framework/Module/PackageInfo.php(156): Magento\Framework\Module\PackageInfo->load()
#2 /var/www/html/magento2/lib/internal/Magento/Framework/Module/DependencyChecker.php(124): Magento\Framework\Module\PackageInfo->getRequire('Magento_Store')
```

Looks like something's wrong with Store `composer.json`. Could you, please, describe how you originally installed Magento? Is it via cloning the Git repository or fully via Composer?
And just in case, could you please, check file `app/code/Magento/Store/composer.json` - does it look valid?

**Comment by milanstojanov** [comment]
Thanks @buskamuza. 

I'm running Merchant Beta, and I installed it by cloning Git repository (https://github.com/magento/magento2.git) and then I used Composer to install dependencies. 

app/code/Magento/Store/composer.json looks normal and valid (tested):

{
    "name": "magento/module-store",
    "description": "N/A",
    "require": {
        "php": "~5.5.0|~5.6.0",
        "magento/module-directory": "1.0.0-beta",
        "magento/module-ui": "1.0.0-beta",
        "magento/module-config": "1.0.0-beta",
        "magento/module-media-storage": "1.0.0-beta",
        "magento/framework": "1.0.0-beta",
        "magento/magento-composer-installer": "*"
    },
    "type": "magento2-module",
    "version": "1.0.0-beta",
    "license": [
        "OSL-3.0",
        "AFL-3.0"
    ],
    "extra": {
        "map": [
            [
                "*",
                "Magento/Store"
            ]
        ]
    }
}

Can you or someone else try installing any M2 module? Is it just me or it's a confirmed bug? If it's just me, I will try reinstalling Magento.

**Comment by mazhalai** [comment]
@milanstojanov if you compare (line 6)[https://github.com/magento/magento2/blob/develop/app/code/Magento/Store/composer.json#L6], you catalog is missing. Also is `*` missing on for `magento/magento-composer-installer` ?

**Comment by milanstojanov** [comment]
You are right. And that's because I pasted composer.json from another laptop where I didn't previously pull the latest code from develop. However, pulling the code alone didn't solve the issue.

What helped in my case is "composer update". I completely forgot to that when I pulled the code. :)

So I can confirm that everything is working as expected. Thanks everyone for participating in this thread. I will close this issue.
